minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java

```diff
@@ -333,7 +335,7 @@ public void setMapPortsToHost(boolean mapPortsToHost) {
     }
 
     public void waitForState(final Predicate<State> predicate) {
-        Awaitility.await().atMost(clusterConfig.getTimeout(), TimeUnit.SECONDS).until(() -> {
+        await("Predicate on Mesos Master failed").atMost(clusterConfig.getTimeout(), TimeUnit.SECONDS).until(() -> {
             try {
```

Comment:
I found a better way of doing these non-trivial waits. You use the version of the callback that doesn't return anything (Runnable) and throw AssertionErrors from it. That way when the timeout expires Awaitility will take the message from the AssertionError showing what exactly has failed.